File: company-application/src/main/java/telran/employees/Manager.java
package telran.employees;

import java.time.LocalDate;

public class Manager extends Employee {
    private double factor;

    public Manager(long id, String name, String department, int salary, LocalDate birthDate, double factor) {
        super(id, name, department, salary, birthDate);
        this.factor = factor;
    }

    @Override
    public String toString() {
        return super.toString() + ", Manager{" +
                "factor=" + factor +
                '}';
    }
}

File: Node-Project/company/src/dto/Manager.mjs
import Employee from "./Employee.mjs";
export default class Manager extends Employee {
    static {
        Employee.classMap.Manager = new Manager();
    }
    constructor(id, department, basicSalary, factor, className) {
        super(id, department, basicSalary, className ?? "Manager");
        this.factor = factor;

    }
    getFactor() {
        return this.factor;
    }
    computeSalary() {
        return super.computeSalary() * this.factor;
    }

}